File: Data Structures & Algorithms/generate-parentheses/submission-1.cpp
class Solution {
public:
    void generateHelper(string current, int left, int right, int n, vector<string>& result) {
        
        if (left == n && right == n) {
            result.push_back(current);
            return;
        }
        
       
        if (left < n) {
            generateHelper(current + '(', left + 1, right, n, result);
        }
        
        
        if (right < left) {
            generateHelper(current + ')', left, right + 1, n, result);
        }
    }

    vector<string> generateParenthesis(int n) {
        vector<string> result;
        generateHelper("", 0, 0, n, result);
        return result;
    }
};

File: Data Structures & Algorithms/climbing-stairs/submission-1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        
        if n == 2:
            return 2

        f1 = 1
        f2 = 2

        for i in range(2,n):
            current = f1 + f2
            f1 = f2
            f2 = current
        
        return f2








 
